data/transactions/asset.go

```diff
@@ -321,8 +321,13 @@ func (ct AssetTransferTxnFields) apply(header Header, balances Balances, spec Sp
 	}
 
 	if ct.AssetCloseTo != (basics.Address{}) {
+		creatorAddr, err := balances.GetAssetCreator(ct.XferAsset)
+		if err != nil {
+			return fmt.Errorf("failed to find asset creator: %v", creatorAddr)
+		}
+
 		// Cannot close asset ID allocated by this account; must use destroy.
-		if ct.XferAsset.Creator == source {
+		if creatorAddr == source {
```

Comment:
Perhaps a more direct check is looking for `AssetParams[ct.XferAsset]` in the source account.  I don't feel too strongly either way though.